src/Trace/Propagator/BinaryFormatter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright 2017 OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenCensus\Trace\Propagator;
+
+use OpenCensus\Trace\TraceContext;
+
+/**
+ * This propagator will contains the method for serializaing and deserializing
+ * TraceContext over a binary format.
+ *
+ * See https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md
+ * for the encoding specification.
+ */
+class BinaryFormatter implements FormatterInterface
+{
+    const OPTION_ENABLED = 1;
+
+    /**
+     * Generate a TraceContext object from the Trace Context header
+     *
+     * @param string $header
+     * @return TraceContext
+     */
+    public function deserialize($bin)
+    {
+        $data = unpack('Cversion/Cfield0/H32traceId/Cfield1/H16spanId/Cfield2/Coptions', $bin);
```

Comment:
Doesn't it need to change the format according to the version?

Reply:
It will, but there's no other versions right now.